### ISSUE TITLE
Results and Record annotations are missing when summary annotation is used.

### ISSUE THREAD

**Issue by khrm** [tektoncd/results#721]
# Expected Behavior
Results and record annotations should be present in pipelineruns( PR ) and taskruns (TRRR).

# Actual Behavior
Results and record annotations aren't present. But we can see that PR and TR are present in database.

# Steps to Reproduce the Problem

1. Pass summary annotation during creation with non-string value.
2. Annotations would be missing.

**Comment by khrm** [comment]
We can see the following error:

```
{"level":"error","time":"2024-03-04T08:52:39.761Z","logger":"watcher","caller":"controller/controller.go:566","msg":"Reconcile error","commit":"5619404","knative.dev/traceid":"0f7c6d4d-9485-4f5a-8ec1-66ef76da1f95","knative.dev/key":"koustav-demo/pipelinerun-pull-request-qwrdz","duration":"1.794702ms","error":"error upserting record: error parsing annotation results.tekton.dev/recordSummaryAnnotations: json: cannot unmarshal number into Go value of type string","stacktrace":"knative.dev/pkg/controller.(*Impl).handleErr\n\t/go/src/github.com/tektoncd/results/vendor/knative.dev/pkg/controller/controller.go:566\nknative.dev/pkg/controller.(*Impl).processNextWorkItem\n\t/go/src/github.com/tektoncd/results/vendor/knative.dev/pkg/controller/controller.go:543\nknative.dev/pkg/controller.(*Impl).RunContext.func3\n\t/go/src/github.com/tektoncd/results/vendor/knative.dev/pkg/controller/controller.go:491"}
{"level":"info","time":"2024-03-04T08:52:39.826Z","logger":"watcher","caller":"dynamic/dynamic.go:176","msg":"Skipping resource deletion: deletion is disabled","commit":"5619404","knative.dev/traceid":"1312c842-3c8e-4fcc-bb58-c97a71dd4447","knative.dev/key":"koustav-demo/pipelinerun-pull-request-qwrdz-maven-build","results.tekton.dev/kind":"TaskRun","results.tekton.dev/result":"koustav-demo/results/269c1c77-4f6f-4c25-a00b-b38c5dfe0ba7","results.tekton.dev/record":"koustav-demo/results/269c1c77-4f6f-4c25-a00b-b38c5dfe0ba7/records/c525ec1a-74f9-46d5-8177-a3e0142a1be8","results.tekton.dev/gracePeriod":"0s"}
```

**Comment by khrm** [comment]
Fixed by #722
